pkg/guacanalytics/patchPlanning_test.go

```diff
@@ -0,0 +1,399 @@
+					Justification:  "test justification one",
+					Origin:         "Demo ingestion",
+					Collector:      "Demo ingestion",
+				},
+			},
+			{
+				Pkg: &model.PkgInputSpec{
+					Type:      "top3",
+					Namespace: ptrfrom.String("topns3"),
+					Name:      "toppkg3",
+					Version:   ptrfrom.String("1.19.0"),
+				},
+				DepPkg: &model.PkgInputSpec{
+					Type:      "top",
+					Namespace: ptrfrom.String("topns"),
+					Name:      "toppkg",
+					Version:   ptrfrom.String("1.19.0"),
+				},
+				IsDependency: &model.IsDependencyInputSpec{
+					VersionRange:   "<1.19.0",
+					DependencyType: model.DependencyTypeDirect,
+					Justification:  "test justification one",
+					Origin:         "Demo ingestion",
+					Collector:      "Demo ingestion",
+				},
+			},
+			{
+				Pkg: &model.PkgInputSpec{
+					Type:      "deb",
+					Namespace: ptrfrom.String("ubuntu"),
+					Name:      "dpkg",
+					Version:   ptrfrom.String("1.19.0"),
+				},
+				DepPkg: &model.PkgInputSpec{
+					Type:      "conan2",
+					Namespace: ptrfrom.String("openssl.org2"),
+					Name:      "openssl2",
+					Version:   ptrfrom.String("3.0.3"),
+				},
+				IsDependency: &model.IsDependencyInputSpec{
+					VersionRange:   ">=1.19.0",
+					DependencyType: model.DependencyTypeDirect,
+					Justification:  "test justification one",
+					Origin:         "Demo ingestion",
+					Collector:      "Demo ingestion",
+				},
+			},
+		},
+	}
+)
+
+func ingestTestData(graphInput string, ctx context.Context, client graphql.Client) {
+	logger := logging.FromContext(ctx)
+	switch graphInput {
+	case "isDependency":
+		for _, ingest := range isDepTestData.IsDependency {
+
+			_, err := model.IngestPackage(context.Background(), client, *ingest.Pkg)
+
+			if err != nil {
+				logger.Errorf("Error in ingesting package: %v\n", err)
+			}
+
+			_, err = model.IngestPackage(context.Background(), client, *ingest.DepPkg)
+
+			if err != nil {
+				logger.Errorf("Error in ingesting dependency package: %v\n", err)
+			}
+			_, err = model.IsDependency(context.Background(), client, *ingest.Pkg, *ingest.DepPkg, *ingest.IsDependency)
+
+			if err != nil {
+				logger.Errorf("Error in ingesting isDependency: %v\n", err)
+			}
+		}
+	}
+}
+
+func Test_SearchSubgraphFromVuln(t *testing.T) {
+	server, logger := startTestServer()
+	ctx := logging.WithLogger(context.Background())
+
+	httpClient := http.Client{}
+	gqlclient := graphql.NewClient("http://localhost:9090/query", &httpClient)
+
+	testCases := []struct {
+		name           string
+		startType      string
+		startNamespace *string
+		startName      string
+		stopType       string
+		stopNamespace  *string
+		stopName       string
+		maxDepth       int
+		expectedLen    int
+		graphInput     string
+	}{
+
+		{
+			name:           "1: test case with two dependencies at the same depth, no stopID and no limiting maxDepth",
+			startType:      "deb",
+			startNamespace: ptrfrom.String("ubuntu"),
+			startName:      "dpkg",
```

Comment:
start node is missing a version, since it needs to know which version to start off from.

Reply:
Fixed.